server/infoextractor.js

```diff
@@ -0,0 +1,181 @@
+const URL = require("url");
+const DailyMotionAdapter = require("./services/dailymotion");
+const GoogleDriveAdapter = require("./services/googledrive");
+const VimeoAdapter = require("./services/vimeo");
+const YouTubeAdapter = require("./services/youtube");
+const DirectVideoAdapter = require("./services/direct");
+const storage = require("../storage");
+const Video = require("../common/video");
+const { UnsupportedMimeTypeException, OutOfQuotaException } = require("./exceptions");
+const { getLogger } = require("../logger");
+
+const log = getLogger("infoextract");
+
+const adapters = [
+  new DailyMotionAdapter(),
+  new GoogleDriveAdapter(),
+  new VimeoAdapter(),
+  new YouTubeAdapter(process.env.YOUTUBE_API_KEY),
+  new DirectVideoAdapter(),
+];
+
+function isURL(str) {
+  return URL.parse(str).host != null;
+}
+
+/**
+ * Returns a cached video and an array with property names. The property names indicate which
+ * properties are still missing from the cache. On a cache miss, this function will return an empty
+ * video object.
+ * @param {string} service
+ * @param {string} videoId
+ * @returns {[string, string[]]}
+ */
+async function getCachedVideo(service, videoId) {
+  try {
+    const result = await storage.getVideoInfo(service, videoId);
+    const video = new Video(result);
+    const missingInfo = storage
+      .getVideoInfoFields(video.service)
+      .filter(p => !video.hasOwnProperty(p));
+
+    if (video.mime && !this.isSupportedMimeType(video.mime)) {
+      throw new UnsupportedMimeTypeException(video.mime);
+    }
```

Comment:
Definitely not the right place to throw this exception

Reply:
That's sort of where it was thrown before. I just changed `video.service === "googledrive"` to `video.mime` to make it independent of the service.

https://github.com/dyc3/opentogethertube/blob/fcb210d6021ac73aea1e97039114e2aba5f31f7d/infoextract.js#L152-L161

I'll see if I can find a better place to throw it.